### PR TITLE
Code health - preprocessor directives

### PR DESCRIPTION
**Pull request by MichelZ**
Unifying preprocessor directives to NET8_0_OR_GREATER

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SSPI/ManagedSSPIContextProvider.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿#if NET6_0
```
Comment (MichelZ) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SSPI/ManagedSSPIContextProvider.cs:1]: This is probably a bug, or does this class need removal in NET8?

Comment (ErikEJ) [reply src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SSPI/ManagedSSPIContextProvider.cs:1]: It should be removed in NET 8

Comment (ErikEJ) [reply src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SSPI/ManagedSSPIContextProvider.cs:1]: That's why I marked it

Comment (MichelZ) [reply src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SSPI/ManagedSSPIContextProvider.cs:1]: I think I'm going to remove all these other !NET8_0_OR_GREATER in another PR

**Comment by ErikEJ** [comment]
I have seen someone mention using !NETFRAMEWORK instead, maybe that is better, avoids doing this for each new .NET?

**Comment by MichelZ** [comment]
> I have seen someone mention using !NETFRAMEWORK instead, maybe that is better, avoids doing this for each new .NET?

True, not the only 8+ is supported anyways, that probably makes sense

**Review by benrr101 — APPROVED** [review]
No issues here 🚢

**Review by David-Engel — APPROVED** [review]

**Review by cheenamalhotra — APPROVED** [review]

**Pull request merged**